### ISSUE TITLE
[Feature] New Ray Lite-Head mode for head node.

### ISSUE THREAD

**Issue by wilsonwang371** [ray-project/ray#22744]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

After a previous discussion with Ant Group, we realized the need to separate head node components into isolated environments. Putting head node specific components(GCS, dashboard, ray client server) into a separated node can help improve the head node stability. Therefore, we are thinking of adding a new "lite-head" mode to head node.

### Use case

When "lite-head" node is enabled, only GCS, dashboard, ray client server will be started on the head node. Other components will not be started. To be able to schedule tasks, one or more workers need to connect to this lite-head node.

We have this code in our downstream branch and we can contribute this back to master if this feature can be accepted by the community.

@iycheng @mwtian 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by jovany-wang** [comment]
Vote +1.

It seems like what we'd propose https://github.com/ray-project/ray/pull/20213 .

**Comment by larrylian** [comment]
Vote +1.

**Comment by jovany-wang** [comment]
At high level, I'm not sure whether it's a good idea to run Rayclient/server/ dashboard with a raylet. 

If we start raylet on head and don't schedule any task, users still be confused why there is 101 nodes and I only use 100. It's a huge mental challenge to users.

**Comment by mwtian** [comment]
Ray client server and dashboard job server needs access to the worker API, so currently they need a raylet as well.

Running raylet run in a "passive mode" to hide it from any "active" node list should be possible. Another way is to allow not running client server or dashboard with GCS. It should be beneficial to make Ray internal routing more modular. External routing to Ray will also be a bit more complicated.

**Comment by wilsonwang371** [comment]
> Ray client server and dashboard job server needs access to the worker API, so currently they need a raylet as well.
> 
> Running raylet run in a "passive mode" to hide it from any "active" node list should be possible. Another way is to allow not running client server or dashboard with GCS. It should be beneficial to make Ray internal routing more modular. External routing to Ray will also be a bit more complicated.

One question, will a "passive mode" raylet populate the object store and have a changing memory usage?

**Comment by mwtian** [comment]
> > Ray client server and dashboard job server needs access to the worker API, so currently they need a raylet as well.
> > Running raylet run in a "passive mode" to hide it from any "active" node list should be possible. Another way is to allow not running client server or dashboard with GCS. It should be beneficial to make Ray internal routing more modular. External routing to Ray will also be a bit more complicated.
> 
> One question, will a "passive mode" raylet populate the object store and have a changing memory usage?

Yes, when used by client server or potentially by job service the local object store could be used. After thinking about this a bit more, I think direction of not running client server or dashboard with GCS, and support running them on different Ray nodes would be more feasible as the first step.

**Comment by wilsonwang371** [comment]
> > > Ray client server and dashboard job server needs access to the worker API, so currently they need a raylet as well.
> > > Running raylet run in a "passive mode" to hide it from any "active" node list should be possible. Another way is to allow not running client server or dashboard with GCS. It should be beneficial to make Ray internal routing more modular. External routing to Ray will also be a bit more complicated.
> > 
> > 
> > One question, will a "passive mode" raylet populate the object store and have a changing memory usage?
> 
> Yes, when used by client server or potentially by job service the local object store could be used. After thinking about this a bit more, I think direction of not running client server or dashboard with GCS, and support running them on different Ray nodes would be more feasible as the first step.

Agreed. But running gcs separately may lead us towards a different long term design. so probably now it is time to think about running gcs separately on different node or nodes?

**Comment by wilsonwang371** [comment]
Just plotted one diagram and later ppl can discuss the feasibility of this HA design:

If we run GCS separately from head node(as we discussed in this thread's previous comments). We may finally end up with a Ray HA design like this:
1. Multiple nodes will do election and the winner will be the new head node.
2. GCS component will be external and can work as a separate cluster. Either active-standby HA or active-active HA mode.

![image](https://user-images.githubusercontent.com/3913185/156839014-db284eee-a5bf-4a8a-820e-454390c01069.png)


FYI: @iycheng @mwtian @jovany-wang @larrylian @chenk008

**Comment by hongbo-miao** [comment]
I originally asked at https://stackoverflow.com/q/79468588/2000548

I found enabling Global Control Service (GCS) fault tolerance only makes Ray job history stays after head node after restarting.

However, I found if a job running and Ray head node dies, after it auto restarts, the job still shows "running", but it is not in healthy state any more and eventually will become "failed".

So today this is still no solution that I am aware of to make Ray head node high availability without interrupting the jobs that are running. 😔
